### DISCUSSION TITLE
Help Convert a reflection coefficient in Magnitude Angle format into Impedance

### DISCUSSION THREAD

**Discussion by stan3c** [scikit-rf/scikit-rf#1025] [Q&A]
I have a python code to convert gamma to impedance:
```
import numpy as np
import skrf.network as net
gamma_opt = (0.29614796004509125+0.3039039572306889j)
z_s = net.s2z(np.array([[[gamma_opt]]]))[0,0,0]
```

which gives me the answer
`z_s = (69.75052435645519+51.70498367058532j)`
which is way off compared to online calculator at [https://leleivre.com/rf_gammatoz.html](https://leleivre.com/rf_gammatoz.html)
92+j0.3 ohm.
What am I doing wrong?

**Comment by jhillairet** [comment]
The website https://leleivre.com/rf_gammatoz.html takes Gamma in Mag and Ang in degree, that is, the following values:

```python
import skrf as rf
gamma_opt = (0.29614796004509125+0.3039039572306889j)
rf.complex_2_magnitude(gamma_opt)  # --> 0.4243362221863005
rf.complex_2_degree(gamma_opt)  # --> 45.740537854365876
```

using these values into the online calculator gives  69.75 + 51.70j Ohm for Z0=50 Ohm, which matches scikit-rf result.

Note that you have another way of doing this calculation using the transmission lines functions of scikit-rf:

```python
import skrf as rf
rf.Gamma0_2_zl(Gamma=gamma_opt, z0=50)  # --> array([69.75052436+51.70498367j])
```

**Reply by jhillairet** [reply]
@stan3c , is your issue solved?
